docs(v1.5.1): add v1.5.1 release notes to CHANGELOG (#1469)
openfga/openfga: CHANGELOG.md

## [Unreleased]

## [1.5.1] - 2024-03-19

[Full changelog](https://github.com/openfga/openfga/compare/v1.5.0...v1.5.1)

> ### Added

Reviewer: Please also mention:

- support for modular models https://github.com/openfga/openfga/pull/1443. cc @rhamzeh 
- support for throttling in Check https://github.com/openfga/openfga/pull/1440

> - Added `request_duration_ms` which has `dispatch_count` as a dimension ([#1444](https://github.com/openfga/openfga/pull/1444))

Reviewer: ```suggestion
- Added `request_duration_ms` histogram which has `datastore_query_count` and `dispatch_count` as dimensions ([#1444](https://github.com/openfga/openfga/pull/1444))
```

- Added an option to specify oidc issuer aliases ([#1354](https://github.com/openfga/openfga/pull/1354)) - Thanks @le-yams!

### Fixed

- We will throw 400 error when condition is invalid instead of "Internal Server Error" ([#1420](https://github.com/openfga/openfga/pull/1420))
- Fix model validation which threw error "no entrypoints defined" ([#1422](https://github.com/openfga/openfga/pull/1422))

### Deprecation :warning:

> - `request_duration_by_query_count_ms` will be removed in favour of `request_duration_ms` ([#1450](https://github.com/openfga/openfga/pull/1450))

Reviewer: ```suggestion
- Histogram `request_duration_by_query_count_ms` will be removed in the next release, in favour of `request_duration_ms` ([#1450](https://github.com/openfga/openfga/pull/1450))
```

## [Unreleased]

## [1.5.1] - 2024-03-19

[Full changelog](https://github.com/openfga/openfga/compare/v1.5.0...v1.5.1)

### Added

- Added `request_duration_ms` histogram which has `datastore_query_count` and `dispatch_count` as dimensions ([#1444](https://github.com/openfga/openfga/pull/1444))

> - Added an option to specify oidc issuer aliases ([#1354](https://github.com/openfga/openfga/pull/1354)) - Thanks @le-yams!
> - Support schema version 1.2 for modular models ([#1443](https://github.com/openfga/openfga/pull/1443))
> - Support for throttling dispatches ([#1440](https://github.com/openfga/openfga/pull/1440))

Reviewer: Sorry, I should have said this in my first comment 😅 WDYT if we add more details on how to turn on these features?

```suggestion
- Added new flag `OPENFGA_AUTHN_OIDC_ISSUER_ALIASES` to specify oidc issuer aliases ([#1354](https://github.com/openfga/openfga/pull/1354)) - Thanks @le-yams!
- Added experimental support for modular models via `OPENFGA_EXPERIMENTALS=enable-modular-models` ([#1443](https://github.com/openfga/openfga/pull/1443)). This will enable writing models that are split across multiple files.
- Added support for throttling dispatches ([#1440](https://github.com/openfga/openfga/pull/1440)). This will throttle Check requests that are overly complex. You can turn on this feature via OPENFGA_DISPATCH_THROTTLING_ENABLED and configured via OPENFGA_DISPATCH_THROTTLING_THRESHOLD and OPENFGA_DISPATCH_THROTTLING_FREQUENCY
```

Author: This is definitely much better. We used automated change log generation with just PR name. This is much better

### Fixed

> - We will throw 400 error when condition is invalid instead of "Internal Server Error" ([#1420](https://github.com/openfga/openfga/pull/1420))
> - Fix model validation which threw error "no entrypoints defined" ([#1422](https://github.com/openfga/openfga/pull/1422))

Reviewer: ```suggestion
- Throw HTTP 400 when tuple condition is invalid instead of HTTP 500 ([#1420](https://github.com/openfga/openfga/pull/1420))
- Fix model validation which threw error "no entrypoints defined" ([#1422](https://github.com/openfga/openfga/pull/1422))
```

> 

Reviewer 2: Added dispatch count for list objects - https://github.com/openfga/openfga/pull/1427?

Author: Isn't that something internal and not exposed to customers? Do we need to add that?

## [Unreleased]

## [1.5.1] - 2024-03-19

[Full changelog](https://github.com/openfga/openfga/compare/v1.5.0...v1.5.1)

### Added

> - Added `dispatch_count` histogram support for `ListObjects` and `StreamedListObjects` methods ([#1469](https://github.com/openfga/openfga/pull/1469))

Reviewer: ```suggestion
- Include calls to ListObjects and StreamedListObjects methods in the `dispatch_count` histogram ([#1427](https://github.com/openfga/openfga/pull/1427))
```
